src/libleo/api/readwrite.c

```diff
@@ -0,0 +1,33 @@
+#include "global.h"
+#include "ultra64/leo.h"
+#include "ultra64/leoappli.h"
+#include "ultra64/leodrive.h"
+
+s32 LeoReadWrite(LEOCmd* cmdBlock, s32 direction, u32 LBA, void* vAddr, u32 nLBAs, OSMesgQueue* mq) {
+    if (!__leoActive) {
+        return -1;
+    }
+
+    if (direction == OS_READ) {
+        cmdBlock->header.command = 5;
+    } else {
+        cmdBlock->header.command = 6;
+    }
+
+    cmdBlock->header.reserve1 = 0;
+
+    if (mq != NULL) {
+        cmdBlock->header.control = 0x80;
+    } else {
+        cmdBlock->header.control = 0;
+    }
+
+    cmdBlock->header.reserve3 = 0;
+    cmdBlock->header.post = mq;
+    cmdBlock->data.readWrite.lba = LBA;
+    cmdBlock->data.readWrite.transferBlks = nLBAs;
+    cmdBlock->data.readWrite.buffPtr = vAddr;
+
+    leoCommand(cmdBlock);
+    return 0;
```

Comment:
```suggestion
    return LEO_ERROR_GOOD;
```

Reply:
The documentation for this says "The return value is -1 if Leo Manager has not been created, and 0 if it has" while for the others it usually says "LEO_ERROR_GOOD" instead of 0. Should we assume they meant LEO_ERROR_GOOD for these too? (ditto several others below)

https://ultra64.ca/files/documentation/online-manuals/functions_reference_manual_2.0i/64dd/LeoReadWrite.html

Reply:
I think it makes sense to use LEO_ERROR_GOOD if other functions that return -1 are documented as returning LEO_ERROR_GOOD. If all functions that return -1 don't use LEO_ERROR_GOOD then it shouldn't be used here.